### DISCUSSION TITLE
Suggest: Moving the Accessibility (A11y) item in Advanced Usage to Get Started.

### DISCUSSION THREAD

**Discussion by yamanoku** [react-hook-form/react-hook-form#5444] [Ideas]
https://react-hook-form.com/advanced-usage#AccessibilityA11y
The accessibility section is great! 👏🏼 

I was thinking that the accessibility items currently listed would be better written in the Handle errors section of Get Started than in the Advanced section.

This is because, as I understand it, accessibility is not ensured by the React Hook Form API, but is written to be ensured using WAI-ARIA.

---

Also, I don't want to treat the Web as an advanced thing, because it is supposed to have accessibility (as Tim Berners-Lee said).

> The power of the Web is in its universality. Access by everyone regardless of disability is an essential aspect.
> 
> Tim Berners-Lee, W3C Director and inventor of the World Wide Web
> http://www.w3.org/standards/webdesign/accessibility

**Comment by bluebill1049** [comment]
Thanks for the suggestion! I agree with all. however, I think `get started` should be as **light** as possible, not saying accessibility should not be the light part. 

I am also open to everyone's suggestions from the community.

**Comment by peterjcole** [comment]
+1 on this, came here to raise the same thing. I was quite disappointed to see accessibility under 'advanced usage'. Using the tool in an accessible way should be the default. 

I'd strongly argue for using the accessible example in Get Started and removing the accessibility section from 'Advanced'

Can take inspiration from other libraries like `react-aria` where accessible is the default (but is also a bit different because it's fully built into the hooks) https://react-spectrum.adobe.com/react-aria/useTextField.html

Would it be helpful if I raised an Issue for this @bluebill1049?

**Comment by leapful** [comment]
@yamanoku @peterjcole @bluebill1049 : May be a back link to "https://react-hook-form.com/advanced-usage#AccessibilityA11y" is ok ?

<img width="1792" alt="Screen Shot 2022-09-23 at 17 11 23" src="https://user-images.githubusercontent.com/18494222/191939272-ca7cfcab-1b3c-4db7-af1f-84b59e26f1d4.png">

**Reply by peterjcole** [reply]
@leapful nice idea but personally I'm not keen, it's still suggesting accessibility is an extra thing and that it's OK for people to think about it later or not bother with it. People should think about it now, and so it should be part of the main example

**Reply by yamanoku** [reply]
> it's still suggesting accessibility is an extra thing and that it's OK for people to think about it later or not bother with it. People should think about it now, and so it should be part of the main example

I have the same thought.
I don't want accessibility to be treated as Advanced, but as a *fundamental aspect* of web development.

**Reply by yamanoku** [reply]
Once again, my suggestion is not to treat accessible codes as Advanced, but to incorporate them universally into each code.

I see that https://react-hook-form.com/advanced-usage/#AccessibilityA11y is the code that makes the form accessible for errors.
If that is the case, I think it would be fine to list the accessible code at https://react-hook-form.com/get-started#Handleerrors as well.
That's all I'm saying.

**Reply by bluebill1049** [reply]
> think it would be fine to list the accessible code at https://react-hook-form.com/get-started#Handleerrors as well.

thanks for the feedback, that's been updated and PR welcome as well.

**Reply by bluebill1049** [reply]
thanks @leapful for jumping in and try to provide a solution ❤️

**Reply by peterjcole** [reply]
Awesome, thanks @bluebill1049!

**Reply by yamanoku** [reply]
Thanks @bluebill1049 !

And, Thanks to @peterjcole for agreeing and commenting on my comment over a year ago!
